fe2-android/app/src/test/java/com/github/dedis/popstellar/ui/home/HomeActivityTest.java

```diff
@@ -0,0 +1,87 @@
+package com.github.dedis.popstellar.ui.home;
+
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.withChild;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static com.github.dedis.popstellar.ui.pages.home.HomePageObject.fragmentContainer;
+import static com.github.dedis.popstellar.ui.pages.home.HomePageObject.homeButton;
+import static com.github.dedis.popstellar.ui.pages.home.HomePageObject.connectButton;
+import static com.github.dedis.popstellar.ui.pages.home.HomePageObject.walletButton;
+import static com.github.dedis.popstellar.ui.pages.home.HomePageObject.socialMediaButton;
+import static com.github.dedis.popstellar.ui.pages.home.HomePageObject.launchButton;
+import static com.github.dedis.popstellar.ui.pages.home.HomePageObject.homeFragmentId;
+import static com.github.dedis.popstellar.ui.pages.home.WalletPageObject.walletFragmentId;
+
+
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import dagger.hilt.android.testing.HiltAndroidRule;
+import dagger.hilt.android.testing.HiltAndroidTest;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnit;
+
+
+@HiltAndroidTest
+@RunWith(AndroidJUnit4.class)
+public class HomeActivityTest {
+
+  //  Hilt rule
+  private final HiltAndroidRule hiltAndroidRule = new HiltAndroidRule(this);
+
+  //Setup rule, used to setup things before the activity is started
+  private final TestRule setupRule =
+      new ExternalResource() {
+        @Override
+        protected void before() throws Throwable {
+          hiltAndroidRule.inject();
+        }
+      };
```

Comment:
Could this simply be done in the @Before ?
Is it needed for the other rules ?

In fact, this is not needed at all as `hiltAndroidRule.inject();` only inject the modules defined in the test class. The modules defined in the rest of the classes are handled by `hiltAndroidRule`